File: pueblo-golf-tournament-api/Dto/PersonDto.cs
using pueblo_golf_tournament_api.Base;

namespace pueblo_golf_tournament_api.Dto
{
    public class PersonDto : BaseDto
    {
        public string? FirstName { get; set; }
        public string? MiddleName { get; set; }
        public string? LastName { get; set; }
        public DateTime BirthDate { get; set; }
        public string? ContactNumber { get; set; }
        public string? HomeAddress { get; set; }
        public string? City { get; set; }
        public string? EmailAddress { get; set; }
        public string? Country { get; set; }
    }
    public class PersoRawDto
    {
        public string? FirstName { get; set; }
        public string? MiddleName { get; set; }
        public string? LastName { get; set; }
        public DateTime BirthDate { get; set; }
        public string? ContactNumber { get; set; }
        public string? EmailAddress { get; set; }
        public string? Country { get; set; }
    }

}

File: pueblo_golf_tournament_mobile/lib/dto/model/person-dto.dart
import 'package:pueblo_golf_tournament_mobile/base/base-dto.dart';
import 'package:json_annotation/json_annotation.dart';

part 'person-dto.g.dart';

@JsonSerializable()
class PersonDto extends BaseDto {
  final String firstName;
  final String middleName;
  final String lastName;
  final DateTime birthDate;
  final String contactNumber;
  final String homeAddress;
  final String city;
  final String emailAddress;
  final String country;
  PersonDto({
    required super.id,
    required this.firstName,
    required this.middleName,
    required this.lastName,
    required this.birthDate,
    required this.contactNumber,
    required this.homeAddress,
    required this.city,
    required this.emailAddress,
    required this.country,
  });

  factory PersonDto.fromJson(Map<String, dynamic> json) =>
      _$PersonDtoFromJson(json);
  Map<String, dynamic> toJson() => _$PersonDtoToJson(this);
}
